### ISSUE TITLE
Karpenter for any Kubernetes Cluster

### ISSUE THREAD

**Issue by manish-jangra** [kubernetes-sigs/karpenter#1613]
### Description

**What problem are you trying to solve?**
We want to make use Karpenter in our ROSA && OSD Clusters for smart autoscaling. I haven't found any documenation of Karpenter usage for Kubernetes Clusters except EKS.

**How important is this feature to you?**
Very important, will help us save cloud costs significantly.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by leoryu** [comment]
You need to implemente your provider. 

Actually, karpenter has implemented based kwok provider. 

You can check https://github.com/kubernetes-sigs/karpenter/tree/main/kwok .

**Comment by jwcesign** [comment]
For ROSA && OSD Clusters, they use the ec2 on aws to build their own control plane and data plane?

**Comment by manish-jangra** [comment]
@jwcesign yes, ROSA and OSD make use of ec2 instances for control plane and data plane nodes.

**Comment by manish-jangra** [comment]
@jwcesign did you happen to find anything in this regard?

**Comment by rschalo** [comment]
Have you been able to review https://github.com/kubernetes-sigs/karpenter-provider-cluster-api? There is also a mega issue for cloudproviders: https://github.com/kubernetes-sigs/karpenter/issues/741
